internal/controllers/gateway/route_utils.go

```diff
@@ -82,6 +84,9 @@ const (
 	// https://github.com/kubernetes-sigs/gateway-api/pull/1516
 	// TODO: swap this out with upstream const when released.
 	RouteReasonNoMatchingParent gatewayv1beta1.RouteConditionReason = "NoMatchingParent"
+	// This reason is used with the "Accepted" condition when no hostnames in listeners
+	// could match hostname in the spec of route.
+	RouteReasonNoMatchingListenerHostname gatewayv1beta1.RouteConditionReason = "NoMatchingListenerHostname"
```

Comment:
You might also consider proposing this upstream. This will either guide us into reusing `RouteReasonNoMatchingParent` (and perhaps extending the comment on it) or pushing this change upstream and reusing their `const` - when released). Similarly to what happened to `RouteReasonNoMatchingParent` https://github.com/kubernetes-sigs/gateway-api/pull/1516

Reply:
Added issue: https://github.com/kubernetes-sigs/gateway-api/issues/1561. Waiting for more discussions and PRs. What should be done on this PR before the upstream issue resolved? I do not think leaving this PR open until related upstream issues and discussions resolves would be the best choice.

Reply:
I believe we can reference this (issue link) in comment, stating that this might change given the feedback we receive from the community. I am personally not convinced we need a separate reason for this ( I think `RouteReasonNoMatchingParent` would work as well in here) but let's wait for feedback from community.

Reply:
added comments.